src/backend/src/bitcoin_utils.rs

```diff
@@ -0,0 +1,32 @@
+use std::str::FromStr;
+use ic_cdk::api::management_canister::bitcoin::{BitcoinNetwork};
+
+fn parse_derivation_path(path: &str) -> Vec<Vec<u8>> {
+    path.split('/')
+        .skip(1) // Skip the leading 'm'
+        .map(|s| {
+            if s.ends_with('\'') {
+                // Handle hardened derivation (apostrophe indicates hardened)
+                let index = u32::from_str(&s[..s.len() - 1]).unwrap() + 0x80000000;
+                index.to_be_bytes().to_vec()
+            } else {
+                let index = u32::from_str(s).unwrap();
+                index.to_be_bytes().to_vec()
+            }
+        })
+        .collect()
+}
+
+pub fn network_to_derivation_path(network: BitcoinNetwork) -> Vec<Vec<u8>> {
+    // This is a simplified version of the complete flow to create a derivation path.
+    // The documentation on the complete flow can be found at https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
+    // For the current implementation and to spare computation workload, we are using the default account 0, change 0, and address index 0.
+    let coin_type = match network {
+        BitcoinNetwork::Mainnet => "0",
+        BitcoinNetwork::Testnet => "1",
+        BitcoinNetwork::Regtest => "1",
```

Comment:
both regtest and testnet are equals to 1, that's correct?

Reply:
Theoretically, we could give whatever derivation path we would like in `Regtest` (even empty), since it is simply a local node (no connection whatsoever with the `Mainnet` or `Testnet` chains, so it is not linked to any convention or specific rules, like BIP-44).

I put the same value as `Testnet` to have consistency between Local env and the env where we will test it properly linked to `Testnet`.

Reply:
I think there is a misunderstanding on what is my enum VS what is the enum in the link.

My list is used to mark the `coin_type` in the derivation path (source [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki)). The list of allowed coin types can be found in [SLIP-0044](https://github.com/satoshilabs/slips/blob/master/slip-0044.md) and it shows the following:

| Coin type  | Path component (`coin_type'`) | Symbol  | Coin                              |
| ---------- | ----------------------------- | ------- | --------------------------------- |
| 0          | 0x80000000                    | BTC     | Bitcoin                           |
| 1          | 0x80000001                    |         | Testnet (all coins)               |
| 2          | 0x80000002                    | LTC     | Litecoin                          |
| 3          | 0x80000003                    | DOGE    | Dogecoin                          |
| ...        | ...                   | ...     | ...                          |

meaning that number 3 is assigned to LiteCoin.

What I do for the derivation path is to give the same `coin_type` to `regtest` as the one of `testnet`: for `regtest` (that is the local Bitcoin node) we actually don't really need a real derivation path that conforms to BIP-44, but I would like to have the same behavior as if it was running in the `testnet` Bitcoin chain.

What do you think?

P.S. I am documenting it in the code.

Reply:
Sounds good, thanks.